packages/fourwings-aggregate/src/types/index.ts

```diff
@@ -0,0 +1,76 @@
+export type BBox = [number, number, number, number]
+
+export enum GeomType {
+  point = 'point',
+  rectangle = 'rectangle',
+}
+
+export enum SublayerCombinationMode {
+  none = 'none',
+  // Add all sublayer raw values
+  add = 'add',
+  // Returns a bucket index depending on sublayer with highest value + position on sublayer color ramp
+  max = 'max',
+  // Returns a bucket index depending on a 2D color ramp
+  bivariate = 'bivariate',
+  // Returns raw values that can be decoded with JSON.parse (number or array of numbers). Used for interaction layer
+  literal = 'literal',
+  // Returns raw values as a string in the format AAAABBBBCCCC (where A, B, C, 3 sublayers), and where BBBB is
+  // sublayer 0 + sublayer 1 and CCCC is sublayer 0 + sublayer 1 + sublayer 2. Used for extruded layer.
+  cumulative = 'cumulative',
+}
+
+export enum AggregationOperation {
+  sum = 'sum',
+  avg = 'avg',
+}
+
+export type BaseTileAggregationParams = {
+  // At which frame should we start counting (related to when the time chunk starts
+  quantizeOffset: number
+  // Number of rames to aggregate, ie number of frames between active time start & end
+  delta: number
+  // Which type of geometry to generate
+  geomType: GeomType
+  // When set to true, aggregation will be done on the backend
+  singleFrame: boolean
+  // When set to true, generate a second sourceLayer that contains the raw data
+  interactive: boolean
+  // Sets value breaks for each sublayer. When set, values will return a bucket index instead of the actual value
+  sublayerBreaks?: number[][]
+  // Sets the number of sublayers
+  sublayerCount: number
+  // Defines how sublayers computed values are combined in cell properties
+  sublayerCombinationMode: SublayerCombinationMode
+  // Defines sublayers visibilty, ie aallow toggling sublayers without reloading source
+  sublayerVisibility: boolean[]
```

Comment:
If we move to identify sublayers by ids, I think would be useful to use them here as well.

Reply:
Let me check. Thing is, since we pass these parameters via query params, there's not so much flexibility in nesting. Although I'm think we could simply use `qs` the same way we do for the URL workspace.

Reply:
Ah I misunderstood your comment. I'm not sure we need to move to identify sublayers by id. Using sublayer indices within HeatmapAnimated layers wwith ids should still work fine